monty.qmd

```diff
@@ -7,15 +7,94 @@ source("common.R")
+The most basic method to define a `monty` model is through a simple R function, as shown in this chapter. More advanced examples are covered in the @sec-monty-models in particular in the context of Bayesian statistics. @sec-monty-dsl introduces the monty DSL for more versatile model building using a small probabilistic DSL similar to BUGs. @sec-monty-combine explains how to compose new monty models by combining two existing ones. The last part of this book starting from @sec-inference demonstrates how to incorporate `odin` models into `monty` workflows.
+
+## A simple example
+
+```{r}
+library(monty)
+```
+
+We can define a simple Gaussian [mixture model](https://en.wikipedia.org/wiki/Mixture_model) of two sub-populations using the `monty_model_function()` from the package. The model represents the distribution of a quantity $l$, sampled with a probability $p$ from a normal distribution of mean $m_{1}$ and with a probability $1-p$ from another normal distribution of mean $m_{2}$. Both subpopulations have variance equal to 1. 
+
+To build our `monty` model, we start by defining an R function returning the log-density of our statistical model. This function has for arguments $l$, the quantity of interest and the three parameters $p$, $m_{1}$ and $m_{1}$ defining the two Normally-distributed subpopulations.
+
+```{r}
+fn <- function(l, p, m1, m2) {
+  log(p * dnorm(l, mean = m1) + (1 - p) * dnorm(l, mean = m2))
+}
+```
+
+Assuming that the population is 75% from subpopulation 1 (normally distributed with mean 3) and 25% from subpopulation 2 (also normally distributed but with mean 7), we can build our `monty` model by indicating that the parameters of the subpopulations are fixed at these given values.
+
+```{r}
+mixture_model <- monty_model_function(fn,
+                                       fixed = list(p = 0.75, m1 = 3, m2 = 7),
+                                       allow_multiple_parameters = TRUE)
+```
+
+We have just created a `monty` model. 
+
+```{r}
+mixture_model
+```
+
+We can plot the density of our model for values of $l$ between 0 and 10 and eye-check that we can see the two subpopulations in the correct proportions and with the 3 and 7 as modes.
+
+```{r}
+#l <- matrix(seq(from = 0, to = 10, by = 0.1), 1)
+l <- seq(from = 0, to = 10, by = 0.1)
+#monty_model_density(l_distribution, l)
+plot(l,
+     exp(Vectorize(mixture_model$density)(l)),
+     type = "l",
+     ylab = "density")
+```
+
+## Sampling from our example distribution
+
+We now want to sample from this model, using the `monty_sample()` function. For this we need to tell `monty` which sampler we want to use to explore our distribution. There are a variety of samplers avalaible and you can learn about them in @sec-samplers. One of the simplest is the random walk [Metropolis-Hastings](https://en.wikipedia.org/wiki/Metropolis%E2%80%93Hastings_algorithm) algorithm that should work almost out of the box (though not necesseraly efficiently) in most cases.
```

Comment:
```suggestion
We now want to sample from this model, using the `monty_sample()` function. For this we need to tell `monty` which sampler we want to use to explore our distribution. There are a variety of samplers avaliable and you can learn about them in @sec-samplers. One of the simplest is the random walk [Metropolis-Hastings](https://en.wikipedia.org/wiki/Metropolis%E2%80%93Hastings_algorithm) algorithm that should work almost out of the box (though not necesseraly efficiently) in most cases.
```

Reply:
Good spot - but not still not the right spelling due to malignant typo

Reply:
Corrected though ;)